maze_transformer/training/tokenizer.py

```diff
@@ -60,3 +71,92 @@ def as_tokens(
             )
 
         return tokens
+
+
+class HuggingMazeTokenizer(PreTrainedTokenizer):
+    vocab: dict[str, int]  # map of token_ids to strings
+
+    bos_token: str = SPECIAL_TOKENS["start_path"]
+    eos_token: str = SPECIAL_TOKENS["end_path"]
+    pad_token: str = SPECIAL_TOKENS["padding"]
+    unk_token: str = "<UNK>"
+
+    vocab_size: int = 0
+    additional_special_tokens: list[str] = [
+        x for x in SPECIAL_TOKENS.values() if x not in [SPECIAL_TOKENS["padding"]]
+    ]
+
+    # Overwrite class attributes
+    padding_side = "left"
+    truncation_side = "left"  #! strange choice, but it's what we did in pad_sequence
+
+    name_or_path = "maze_tokenizer"
+
+    def __init__(self, cfg: Union["ConfigHolder", "MazeDatasetConfig"], **kwargs):
```

Comment:
We are doing a lot of work in this constructor. My instinct says that we should simplify the constructor to just call super and do any assignment, and any additional work to a factory method. Maybe my mindset is too OOP. I'm not sure. @luciaquirke @valedan interested in your thoughts here too.